Armadillo/src/main/java/com/scribd/armadillo/download/MaxAgeCacheEvictor.kt

```diff
@@ -71,29 +99,50 @@ internal class MaxAgeCacheEvictor(private val maxBytes: Long,
 
     private fun evictCache(cache: Cache, requiredSpace: Long) {
 
-        var oldestSpan = oldestCacheSpan()
+        // remove expired content
+        while (expirations.peekLast()?.expiration?.isExpired() == true) {
```

Comment:
so the `CacheSpanExpiration` is static and it can't access member variables like `clock`. I could add a method on the data class that accepts current time as an argument. That would end up being quite long to have in the while loop

ex. `while (expirations.peekLast()?.isExpired(clock.currentTimeMillis()) == true) {`

what do you think?

Reply:
I can think of a couple possibilities:

1) You could make the extension method on `CacheSpanExpiration` instead
2) Make it an inner class so it has access to the clock - it's only used here, so I think that's fine to do
3) Pass the clock into the method

I think I like option #2 the best. What do you think?

Reply:
We discussed offline and agreed on #1 because our data class is still static and we can get access to class methods with an extension function.